### PR TITLE
Added possibility to simply create own combiners

### PR DESCRIPTION
**Pull request by Skazu**

### CODE REVIEW

File: darwin/abc.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
     @abc.abstractmethod
     def __call__(self, population):
         """Mutate the given population iterable"""
+
+
+class BaseCombiner(BaseMutator):
+    NUM_PARENTS = 2
+    KEEP_PARENTS = False
+
+    def __call__(self, population: list):
```
Comment (sighalt) [inline darwin/abc.py]: Please use typing.Iterable instead of list

File: darwin/mutators.py [hunk 2]
```diff
@@ -61,3 +61,15 @@ def __call__(self, population):
 
         for history_list in history_lists:
             history_list.apply_changes(population)
+
+
+
+class SimpleCombiner(BaseCombiner):
+    def __init__(self, combiner, keep_parents=False, num_parents=2):
```
Comment (sighalt) [inline darwin/mutators.py]: Please stick with the convention to start the names of parameters, indicating a size, with 'n_'

File: darwin/abc.py [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
+
+            if not isinstance(children, (list, tuple)):
+                children = [children]
+
+            combined_children += children
+
+        if not self.KEEP_PARENTS:
+            population.clear()
+
+        for child in combined_children:
```
Comment (sighalt) [inline darwin/abc.py]: Use `population.extend(combined_children)` for performance

File: darwin/abc.py [hunk 4]
```diff
@@ -1,8 +1,35 @@
...
+    NUM_PARENTS = 2
+    KEEP_PARENTS = False
+
+    def __call__(self, population: list):
+        combined_children = list()
+
+        for parents in itertools.combinations(population, self.NUM_PARENTS):
+            children = self.combine(parents)
+
+            if not isinstance(children, (list, tuple)):
```
Comment (sighalt) [inline darwin/abc.py]: It seems plausible to me that users will use `yield` to generate the offspring. Please check if the instance is of type `collections.Iterable`.

**Review by sighalt — CHANGES_REQUESTED** [review]
Thanks for your effort. I left you with some comments.

**Comment by Skazu** [comment]
I've worked on your suggestions and changed a few more parts of the code, please review again.

**Review by sighalt — APPROVED** [review]
Thanks.

**Pull request merged**